push_notifications/api/rest_framework.py

```diff
@@ -125,6 +128,22 @@ def has_object_permission(self, request, view, obj):
 class DeviceViewSetMixin(object):
 	lookup_field = "registration_id"
 
+	def create(self, request, *args, **kwargs):
+		serializer = None
+		if SETTINGS.get('AUTO_UPDATE_MATCHING_REG_ID') and 'registration_id' in request.data:
+			instance = self.queryset.model.objects.filter(
+				registration_id=request.data['registration_id']
+			).first()
+			if instance:
+				serializer = self.get_serializer(instance, data=request.data)
+		if not serializer:
+			serializer = self.get_serializer(data=request.data)
+
+		serializer.is_valid(raise_exception=True)
+		self.perform_create(serializer)
+		headers = self.get_success_headers(serializer.data)
+		return Response(serializer.data, status=status.HTTP_201_CREATED, headers=headers)
```

Comment:
Looks good, but in the case of an update the http status code should be 200 or 204